### ISSUE TITLE
Add `ip-forward-policy` to deamon configuration

### ISSUE THREAD

**Issue by wclr** [moby/moby#30093]
Due to this issue https://github.com/docker/docker/issues/14041 this [PR](https://github.com/docker/libnetwork/pull/1526) introduced  default forward `DROP` policy instead of `ALLOW`, this doesn't allow to on docker for windows (and I assume for mac) to access internal docker network from developer's machine **which is really useful in development mode** with docker, the issue: https://github.com/docker/for-win/issues/221

If to change this policy to ALLOW in MobyLinuxVM, forwarding routing start to work. I propose to add a param that could override default behaviour of this policy.

**Comment by justincormack** [comment]
It doesn't apply to Mac as there is currently no route to the VM IP address.

**Comment by whitecolor** [comment]
On windows there is too no route, one should add this to routing table, though it is onetime action.

**Comment by justincormack** [comment]
@whitecolor on Mac you cannot add a route, there is no shared network stack.

**Comment by whitecolor** [comment]
@justincormack have to be another way?

**Comment by FdSMD** [comment]
@whitecolor 
Following up from https://github.com/docker/for-win/issues/221 I just have one big pile of kudo for you.

Thanks for all the efforts put in this, because, guess what, I'm facing the exact same issue (win 10dev machine / linux based containers / SOA). I know it would be just simpler to have nux dev machine or that I could run a VM to run docker but ... one cannot always choose the company OS and running a docker based dev environment inside vm also comes with a productivity cost.

So I also think that and ip-forward-policy config set to DROP by default with all the necessary warnings about changing it is indeed a must, even if this does only address this specific use case (eg not mac)
Note that I'm not saying that a feature was broken here, I'm aware of the specificity of the matter. I'm just saying that it seems to me that such a config entry would not represent much work or risk to implement and would be a life saver for at least some (well, the insane ones using win dev machines to work with nux based infra).

**Comment by whitecolor** [comment]
The problems is that docker may drop even such opportunity (changing the policy) to solve the issue if I get it right from the comment https://github.com/docker/for-win/issues/221#issuecomment-272143387, I wonder why they don't see this as an important dev use case.

**Comment by FdSMD** [comment]
OK, we have to go politics on that then.

Elect my request because : 
**why would you have put so much efforts into docker for windows otherwise ?**
I mean, that's cool to see MS interest into MS container but I do think that the life saver use case is to be able to run nux container efficiently on windows. So a part from the adventurous ones, who is going to promote docker on windows the most ?

**the pro/cons ratio**
I fully understand why the policy was changed, and it is a good thing. But, this breaks something that used to work (again, I'm not advocating that anything was broken since it was indeed kind of a hack anyway but ...) and that does still work for the lucky ones who can use nux dev machines.
In other words, if all it takes to make it possible to have the same dev comfort on nux and windows (we can see the add route as insignificant) is to make one property configurable, it looks to me that the pros are winning.
It's not even a security risk as it would of course comes with the default DROP value.

Please docker, I worked hard to convince people at my company to use docker, really I did ^^

**Comment by whitecolor** [comment]
@justincormack 

At least on mac it works with Docker ToolBox (with Virtual Box VM), no problems with adding needed route to Docker machine VM, and it works without need of policies changes.

Probably the same will be for Windows Docker Toolbox.

**Comment by thaJeztah** [comment]
Let me close this ticket for now, as it looks like it went stale.
